lib/galaxy/webapps/galaxy/api/sanitize_allow.py

```diff
@@ -0,0 +1,84 @@
+"""
+API operations allowing clients to retrieve and modify the HTML sanitization allow list.
+"""
+import logging
+
+from galaxy import web
+from galaxy.webapps.base.controller import BaseAPIController
+
+log = logging.getLogger(__name__)
+
+
+class SanitizeAllowController(BaseAPIController):
+
+    @web.require_admin
+    @web.expose_api
+    def index(self, trans, **kwd):
+        """
+        GET /api/sanitize_allow
+        Return an object showing the current state of the toolbox and allow list.
+        """
+        return {'status': 'done', 'message': 'Tool allow list loaded.', 'data': self._generate_allowlist(trans)}
+
+    @web.require_admin
+    @web.expose_api
+    def create(self, trans, tool_id, **kwd):
+        """
+        PUT /api/sanitize_allow
+        Add a new tool specification to the allow list.
+        """
+        if tool_id not in trans.app.config.sanitize_allowlist:
+            trans.app.config.sanitize_allowlist.append(tool_id)
+            self._save_allowlist(trans)
+        return {'status': 'done', 'message': '%s added to allowlist.' % tool_id, 'data': self._generate_allowlist(trans)}
+
+    @web.require_admin
+    @web.expose_api
+    def delete(self, trans, tool_id, **kwd):
+        """
+        DELETE /api/sanitize_allow
+        Add a new tool specification to the allow list.
+        """
+        if tool_id in trans.app.config.sanitize_allowlist:
+            trans.app.config.sanitize_allowlist.remove(tool_id)
+            self._save_allowlist(trans)
+        return {'status': 'done', 'message': '%s removed from allowlist.' % tool_id, 'data': self._generate_allowlist(trans)}
+
+    def _save_allowlist(self, trans):
+        new_allowlist = sorted([tid for tid in trans.app.config.sanitize_allowlist])
+        trans.app.config.sanitize_allowlist = new_allowlist
+        with open(trans.app.config.sanitize_allowlist_file, 'wt') as f:
+            f.write("\n".join(new_allowlist))
```

Comment:
We need to send a control task, otherwise other processes won't load the new allowlist
```suggestion
            f.write("\n".join(new_allowlist))
        trans.app.queue_worker.send_control_task('reload_sanitize_allowlist', noop_self=True)
```

Reply:
```suggestion
            f.write("\n".join(trans.app.config.sanitize_allowlist))
```